sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/implementation/OkHttpFluxRequestBody.java

```diff
@@ -83,15 +83,23 @@ public void writeTo(BufferedSink bufferedSink) throws IOException {
                 }
             }, 1, 1).then();
 
-            // The blocking happens on OkHttp thread pool.
-            if (callTimeoutMillis > 0) {
-                /*
-                 * Default call timeout (in milliseconds). By default there is no timeout for complete calls, but
-                 * there is for the connection, write, and read actions within a call.
-                 */
-                requestSendMono.block(Duration.ofMillis(callTimeoutMillis));
-            } else {
-                requestSendMono.block();
+            try {
+                // The blocking happens on OkHttp thread pool.
+                if (callTimeoutMillis > 0) {
+                    /*
+                     * Default call timeout (in milliseconds). By default there is no timeout for complete calls, but
+                     * there is for the connection, write, and read actions within a call.
+                     */
+                    requestSendMono.block(Duration.ofMillis(callTimeoutMillis));
+                } else {
+                    requestSendMono.block();
+                }
+            } catch (RuntimeException e) {
+                if (e.getCause() instanceof IOException) {
+                    throw (IOException) e.getCause();
+                } else {
+                    throw e;
+                }
```

Comment:
Will need to add a Checkstyle suppression for this. You can run `python .\eng\scripts\linting_suppression_generator.py -p .\sdk\core\azure-core-http-okhttp\` to have this be done automatically.